homeassistant/components/soma/config_flow.py

```diff
@@ -39,12 +39,18 @@ def __init__(self):
         """Finish config flow."""
         api = SomaApi(user_input["host"], user_input["port"])
         try:
-            await self.hass.async_add_executor_job(api.list_devices)
+            result = await self.hass.async_add_executor_job(api.list_devices)
             _LOGGER.info("Successfully set up Soma Connect")
-            return self.async_create_entry(
-                title="Soma Connect",
-                data={"host": user_input["host"], "port": user_input["port"]},
-            )
+            if result["result"] == "success":
```

Comment:
If `result` is missing the `"result"` key there will be a KeyError. Could that happen?

I'm thinking that's what the stacktrace in the issue describes.

Reply:
That seems to be what is happening, yes. I can add a handler for that case too but that will not fix the underlying issue. I will add that exception handler to this PR just to make it pretty though.

The issue seems to be that our hub takes too long to respond and HA times out before the response comes back. I'm investigating why it would take so long. Due to our hub integrating with multiple systems we decided to only respond to API requests after the BLE communication has finished there will always be a delay in the response but it should ideally be less than a second.